include/nil/blueprint/handle_component.hpp

```diff
@@ -61,11 +61,18 @@
 #include <nil/blueprint/asserts.hpp>
 #include <nil/blueprint/stack.hpp>
 #include <nil/blueprint/policy/policy_manager.hpp>
+#include <typeinfo>
 
 
 namespace nil {
     namespace blueprint {
 
+        std::map<std::string, int> components_frequency;
```

Comment:
1. I suggest use map<sting, statistic>
```
struct statistic {
int components_frequency;
int rows_used;
int gates_used
}
```
2. Global variable may be not better design solution. I guggest to keep map in parser instance and pass as reference.